Making sure FindMauiContext fallbacks to default for the title icon i…

atoghyani opened the pull request:
### Description of Change

Passing a true value to FindMauiContext to make sure context is not null for Title Icon on iOS.
The ImageSource does not have a handler or a parent, so the method will return a null value and causes a crash.

### Issues Fixed

<!-- Please make sure that there is a bug logged for the issue being fixed. The bug should describe the problem and how to reproduce it. -->

Fixes #21518


<!--
Are you targeting main? All PRs should target the main branch unless otherwise noted.
-->

atoghyani committed:
Making sure FindMauiContext fallbacks to default for the title icon image

rmarinho commented:
/azp run

rmarinho reviewed (dismissed):
Hi @atoghyani we add a test for this, could be a UITest or Device test I think, if it crashes  with the fix could be easy to reproduce on a test.

atoghyani committed:
Update NavigationPageTests.cs

Adding SettingTitleIconImageSourceDoesntCrash test to ensure the app does not crash when setting the Icon ImageSource

atoghyani commented:
> Hi @atoghyani we add a test for this, could be a UITest or Device test I think, if it crashes with the fix could be easy to reproduce on a test.

Added a Device test for this:

Before:

![Simulator Screenshot - iPhone 15 Pro Max - 2024-04-01 at 14 13 54](https://github.com/dotnet/maui/assets/25441400/cca7af69-0289-4a61-80a0-2085cdb0950c)
```    at Microsoft.Maui.ImageSourceExtensions.GetPlatformImageAsync(IImageSource imageSource, IMauiContext mauiContext) in /Users/ariantoghyani/Projects/maui/src/Core/src/ImageSources/ImageSourceExtensions.cs:line 40
   at Microsoft.Maui.ImageSourceExtensions.LoadImage(IImageSource source, IMauiContext mauiContext, Action`1 finished) in /Users/ariantoghyani/Projects/maui/src/Core/src/ImageSources/ImageSourceExtensions.cs:line 25
   at Microsoft.Maui.Controls.Handlers.Compatibility.NavigationRenderer.ParentingViewController.UpdateTitleImage(Container titleViewContainer, ImageSource titleIcon) in /Users/ariantoghyani/Projects/maui/src/Controls/src/Core/Compatibility/Handlers/NavigationPage/iOS/NavigationRenderer.cs:line 1454
   at Microsoft.Maui.Controls.Handlers.Compatibility.NavigationRenderer.ParentingViewController.UpdateTitleArea(Page page) in /Users/ariantoghyani/Projects/maui/src/Controls/src/Core/Compatibility/Handlers/NavigationPage/iOS/NavigationRenderer.cs:line 1432
   at Microsoft.Maui.Controls.Handlers.Compatibility.NavigationRenderer.ParentingViewController.HandleChildPropertyChanged(Object sender, PropertyChangedEventArgs e) in /Users/ariantoghyani/Projects/maui/src/Controls/src/Core/Compatibility/Handlers/NavigationPage/iOS/NavigationRenderer.cs:line 1284
   at Microsoft.Maui.Controls.Element.OnPropertyChanged(String propertyName) in /Users/ariantoghyani/Projects/maui/src/Controls/src/Core/Element/Element.cs:line 571
   at Microsoft.Maui.Controls.BindableObject.SetValueActual(BindableProperty property, BindablePropertyContext context, Object value, Boolean currentlyApplying, SetValueFlags attributes, SetterSpecificity specificity, Boolean silent) in /Users/ariantoghyani/Projects/maui/src/Controls/src/Core/BindableObject.cs:line 641
   at Microsoft.Maui.Controls.BindableObject.SetValueCore(BindableProperty property, Object value, SetValueFlags attributes, SetValuePrivateFlags privateAttributes, SetterSpecificity specificity) in /Users/ariantoghyani/Projects/maui/src/Controls/src/Core/BindableObject.cs:line 569
   at Microsoft.Maui.Controls.BindableObject.SetValue(BindableProperty property, Object value) in /Users/ariantoghyani/Projects/maui/src/Controls/src/Core/BindableObject.cs:line 474
   at Microsoft.Maui.Controls.NavigationPage.SetTitleIconImageSource(BindableObject bindable, ImageSource value) in /Users/ariantoghyani/Projects/maui/src/Controls/src/Core/NavigationPage/NavigationPage.cs:line 348
   at Microsoft.Maui.DeviceTests.NavigationPageTests.<>c__DisplayClass2_0.<<SettingTitleIconImageSourceDoesntCrash>b__0>d.MoveNext() in /Users/ariantoghyani/Projects/maui/src/Controls/tests/DeviceTests/Elements/NavigationPage/NavigationPageTests.cs:line 78
--- End of stack trace from previous location ---
   at Microsoft.Maui.DeviceTests.ControlsHandlerTestBase.<>c__DisplayClass16_1`1.<<CreateHandlerAndAddToWindow>b__1>d[[Microsoft.Maui.DeviceTests.Stubs.WindowHandlerStub, Microsoft.Maui.Controls.DeviceTests, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].MoveNext() in /Users/ariantoghyani/Projects/maui/src/Controls/tests/DeviceTests/ControlsHandlerTestBase.cs:line 252
--- End of stack trace from previous location ---
   at Microsoft.Maui.DeviceTests.ControlsHandlerTestBase.<>c__DisplayClass30_0`1.<<SetupWindowForTests>b__0>d[[Microsoft.Maui.DeviceTests.Stubs.WindowHandlerStub, Microsoft.Maui.Controls.DeviceTests, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].MoveNext() in /Users/ariantoghyani/Projects/maui/src/Controls/tests/DeviceTests/ControlsHandlerTestBase.iOS.cs:line 29
--- End of stack trace from previous location ---
   at Microsoft.Maui.DeviceTests.ControlsHandlerTestBase.<>c__DisplayClass30_0`1.<<SetupWindowForTests>b__0>d[[Microsoft.Maui.DeviceTests.Stubs.WindowHandlerStub, Microsoft.Maui.Controls.DeviceTests, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].MoveNext() in /Users/ariantoghyani/Projects/maui/src/Controls/tests/DeviceTests/ControlsHandlerTestBase.iOS.cs:line 86
--- End of stack trace from previous location ---
   at Microsoft.Maui.Dispatching.DispatcherExtensions.<>c__DisplayClass3_0.<<DispatchAsync>b__0>d.MoveNext() in /Users/ariantoghyani/Projects/maui/src/Core/src/Dispatching/DispatcherExtensions.cs:line 88
--- End of stack trace from previous location ---
   at Microsoft.Maui.Dispatching.DispatcherExtensions.<>c__DisplayClass2_0`1.<<DispatchAsync>b__0>d[[System.Boolean, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext() in /Users/ariantoghyani/Projects/maui/src/Core/src/Dispatching/DispatcherExtensions.cs:line 67
--- End of stack trace from previous location ---
   at Microsoft.Maui.DeviceTests.ControlsHandlerTestBase.<>c__DisplayClass16_0`1.<<CreateHandlerAndAddToWindow>b__0>d[[Microsoft.Maui.DeviceTests.Stubs.WindowHandlerStub, Microsoft.Maui.Controls.DeviceTests, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].MoveNext() in /Users/ariantoghyani/Projects/maui/src/Controls/tests/DeviceTests/ControlsHandlerTestBase.cs:line 167
--- End of stack trace from previous location ---
   at Microsoft.Maui.Dispatching.DispatcherExtensions.<>c__DisplayClass3_0.<<DispatchAsync>b__0>d.MoveNext() in /Users/ariantoghyani/Projects/maui/src/Core/src/Dispatching/DispatcherExtensions.cs:line 88
--- End of stack trace from previous location ---
   at Microsoft.Maui.Dispatching.DispatcherExtensions.<>c__DisplayClass2_0`1.<<DispatchAsync>b__0>d[[System.Boolean, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext() in /Users/ariantoghyani/Projects/maui/src/Core/src/Dispatching/DispatcherExtensions.cs:line 67
--- End of stack trace from previous location ---
   at Microsoft.Maui.DeviceTests.NavigationPageTests.SettingTitleIconImageSourceDoesntCrash() in /Users/ariantoghyani/Projects/maui/src/Controls/tests/DeviceTests/Elements/NavigationPage/NavigationPageTests.cs:line 74
--- End of stack trace from previous location ---
``` 

After:

![Simulator Screenshot - iPhone 15 Pro Max - 2024-04-01 at 14 22 26](https://github.com/dotnet/maui/assets/25441400/91ce011d-5f30-438f-b3fc-c9cfd61d9e89)

atoghyani commented:
@dotnet-policy-service agree

rmarinho commented:
/azp run

jsuarezruiz commented on src/Controls/src/Core/Compatibility/Handlers/NavigationPage/iOS/NavigationRenderer.cs:
```diff
@@ -1451,7 +1451,7 @@ void UpdateTitleImage(Container titleViewContainer, ImageSource titleIcon)
 				else
 				{
-					titleIcon.LoadImage(titleIcon.FindMauiContext(), result =>
+					titleIcon.LoadImage(titleIcon.FindMauiContext(true), result =>
```
Seems similar to https://github.com/dotnet/maui/pull/14634, @PureWeen Thoughts?

PureWeen replied:
I think you can just use the `MauiContext` that's on this class and call that good

atoghyani committed:
Update NavigationRenderer.cs

Updating ParentingViewController to use the NavigationRender's MauiContext for loading the title icon image

atoghyani committed:
Update NavigationRenderer.cs

Adding missing line

PureWeen committed:
Update NavigationRenderer.cs

PureWeen committed:
Update NavigationRenderer.cs

PureWeen commented:
/rebase

PureWeen commented:
/azp run

PureWeen commented:
/azp run

PureWeen committed:
Update NavigationRenderer.cs

PureWeen committed:
Update NavigationRenderer.cs

PureWeen commented:
/azp run

pratik088 commented:
How to resolve this issue? I am still having this issue currently on my maui iOS app.
